File: CheckOutApp/Python/check_out_app.py
# Global lists to store cart items
# FUNCTION 1: Collect items in a loop until cashier says no 
# FUNCTION 2: Print store header, item list and totals, return bill total 
# Print store header
# Print column headers
# Loop through all items and print each row
# Calculate discount and VAT
# Print summary
# FUNCTION 3: Print Invoice that shows before customer pays 
# FUNCTION 4: Print Receipt that shows after customer pays 
# Create Main 


items       = []                            
quantities  = []                            
unit_prices = []                           



def collect_items():
    add_more = "yes"                       

    while add_more == "yes":                
        items.append(input("What did the customer buy? "))          
        quantities.append(int(input("How many pieces? ")))           
        unit_prices.append(float(input("How much per unit? ")))     

        add_more = input("Add more items? (yes/no) ").strip().lower() 



def print_store_details(customer, cashier, discount_percent):

    
    print("\n=========================================================")
    print("SEMICOLON STORES")                               
    print("MAIN BRANCH")                                    
    print("LOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS.") 
    print("TEL: 03264263736")                             
    print("Cashier: " + cashier)                            
    print("Customer Name: " + customer)                    
    print("=========================================================")

    
    print(f"{'ITEM':>15} {'QTY':>10} {'PRICE':>10} {'TOTAL(NGN)':>16}")
    print("---------------------------------------------------------")

    
    sub_total = 0.0                                         
    for i in range(len(items)):                            
        total_per_item = quantities[i] * unit_prices[i]    
        sub_total += total_per_item                         
        print(f"{items[i]:>15} {quantities[i]:>10} {unit_prices[i]:>10.2f} {total_per_item:>16.2f}")  

    
    discount_amount = sub_total * (discount_percent / 100)  
    vat_amount      = sub_total * 0.075                     

    
    print("---------------------------------------------------------")
    print(f"{'Sub total':>38}: {sub_total:>16.2f}")         
    print(f"{'Discount':>38}: {discount_amount:>16.2f}")    
    print(f"{'VAT @ 7.5%':>38}: {vat_amount:>16.2f}")       
    print("=========================================================")

    return sub_total - discount_amount + vat_amount         



def print_invoice(customer, cashier, discount_percent):
    bill_total = print_store_details(customer, cashier, discount_percent)  

    print(f"{'Bill Total':>38}: {bill_total:>16.2f}")        
    print("=========================================================")
    print(f"THIS IS NOT A RECEIPT, KINDLY PAY {bill_total:.2f}")  
    print("=========================================================")



def print_receipt(customer, cashier, discount_percent, amount_paid):
    bill_total = print_store_details(customer, cashier, discount_percent)  
    balance    = amount_paid - bill_total                   

    print(f"{'Bill Total':>38}: {bill_total:>16.2f}")        
    print(f"{'Amount Paid':>38}: {amount_paid:>16.2f}")     
    print(f"{'Balance':>38}: {balance:>16.2f}")              
    print("=========================================================")
    print("         THANK YOU FOR YOUR PATRONAGE        ")   
    print("=========================================================")



customer_name    = input("What is the customer's name? ")              

collect_items()                                                         

cashier_name     = input("What is your name? ")                         
discount_percent = float(input("How much discount will be applied? "))  

print_invoice(customer_name, cashier_name, discount_percent)           

amount_paid = float(input("How much did the customer pay? "))           

print_receipt(customer_name, cashier_name, discount_percent, amount_paid)  

File: CheckOutApp/Java/CheckOutApp.java
import java.util.Scanner;

public class CheckOutApp {   

        static int maxItems  = 100;                      
        static String[] items = new String[maxItems];    
        static int[] quantities = new int[maxItems];       
        static double[] unitPrices = new double[maxItems];    
        static int itemCount  = 0;                  

    public static void main(String[] args) {
        Scanner inputCollector = new Scanner(System.in);        

        System.out.print("What is the customer's name? ");
        String customerName = inputCollector.nextLine();               

        collectItems(inputCollector);                                   

        System.out.print("What is your name? ");
        String cashierName = inputCollector.nextLine();                 

        System.out.print("How much discount will be applied? ");
        double discountPercent = inputCollector.nextDouble();           

        printInvoice(customerName, cashierName, discountPercent);   

        System.out.print("How much did the customer pay? ");
        double amountPaid = inputCollector.nextDouble();                

        printReceipt(customerName, cashierName, discountPercent, amountPaid);                                        
    }


    public static void collectItems(Scanner inputCollector) {
        String addMore = "yes";                            

        while (addMore.equals("yes") && itemCount < maxItems) { 

            System.out.print("What did the customer buy? ");
            items[itemCount] = inputCollector.nextLine();               

            System.out.print("How many pieces? ");
            quantities[itemCount] = inputCollector.nextInt();  
                    
            System.out.print("How much per unit? ");
            unitPrices[itemCount] = inputCollector.nextDouble();        
            inputCollector.nextLine();                                  

            itemCount++;                                   

            System.out.print("Add more items? (yes/no) ");
            addMore = inputCollector.nextLine().trim().toLowerCase();   
        }
    }

   
    public static double printStoreDetails(String customer, String cashier, double discountPercent) {

        
        System.out.println("\n=========================================================");
        System.out.println("SEMICOLON STORES");                           
        System.out.println("MAIN BRANCH");                                
        System.out.println("LOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS."); 
        System.out.println("TEL: 03264263736");                             
        System.out.println("Cashier: " + cashier);                         
        System.out.println("Customer Name: " + customer);                  
        System.out.println("============================================================");

        System.out.printf("%15s %10s %10s %16s\n", "ITEM", "QTY", "PRICE", "TOTAL(NGN)");
        System.out.println("---------------------------------------------------------");

        double subTotal = 0;                                
        for (int count = 0; count < itemCount; count++) {             
            double totalPerItem = quantities[count] * unitPrices[count];   
            subTotal += totalPerItem;                      
            System.out.printf("%15s %10d %10.2f %16.2f\n", items[count], quantities[count], unitPrices[count], totalPerItem); 
        }

        
        double discountAmount = subTotal * (discountPercent / 100); 
        double vatAmount      = subTotal * 0.075;                 

        System.out.println("---------------------------------------------------------");
        System.out.printf("%38s: %16.2f\n", "Sub total",  subTotal);       
        System.out.printf("%38s: %16.2f\n", "Discount",   discountAmount); 
        System.out.printf("%38s: %16.2f\n", "VAT @ 7.5%", vatAmount);      
        System.out.println("=========================================================");

        return subTotal - discountAmount + vatAmount;      
    }

   
    public static void printInvoice(String customer, String cashier, double discountPercent) {
        double billTotal = printStoreDetails(customer, cashier, discountPercent);

        System.out.printf("%38s: %16.2f\n", "Bill Total", billTotal);      
        System.out.println("=========================================================");
        System.out.printf("THIS IS NOT A RECEIPT, KINDLY PAY %.2f\n", billTotal); 
        System.out.println("=========================================================");
    }

    
    public static void printReceipt(String customer, String cashier, double discountPercent, double amountPaid) {
        double billTotal = printStoreDetails(customer, cashier, discountPercent); 
        double balance   = amountPaid - billTotal;         

        System.out.printf("%38s: %16.2f\n", "Bill Total",  billTotal);     
        System.out.printf("%38s: %16.2f\n", "Amount Paid", amountPaid);   
        System.out.printf("%38s: %16.2f\n", "Balance",     balance);      
        System.out.println("=========================================================");
        System.out.println("         THANK YOU FOR YOUR PATRONAGE        ");           
        System.out.println("=========================================================");
    }
}
